src/nal/sps.rs

```diff
@@ -1090,4 +1094,532 @@ mod test {
+            frame_cropping: Some(FrameCropping{
+                bottom_offset: 4,
+                ..FrameCropping::default()
+            }),
+            vui_parameters: Some(VuiParameters{
+                timing_info: Some(TimingInfo{
+                    num_units_in_tick: 1,
+                    time_scale:      60,
+                    fixed_frame_rate_flag: false,
+                }),
+                bitstream_restrictions: Some(BitstreamRestrictions{
+                    motion_vectors_over_pic_boundaries_flag: true,
+                    log2_max_mv_length_horizontal: 11,
+                    log2_max_mv_length_vertical: 11,
+                    max_dec_frame_buffering: 3,
+                    ..BitstreamRestrictions::default()
+                }),
+                ..VuiParameters::default()
+            }),
+        },
+        1920,
+        1080,
+        30.0; "1920x1080 baseline"
+    )]
+    #[test_case(
+        vec![
+            0x67, 0x64, 0x00, 0x28, 0xac, 0xd9, 0x40, 0x78,
+            0x02, 0x27, 0xe5, 0x84, 0x00, 0x00, 0x03, 0x00,
+            0x04, 0x00, 0x00, 0x03, 0x00, 0xf0, 0x3c, 0x60,
+            0xc6, 0x58,
+        ],
+        SeqParameterSet{
+            profile_idc: ProfileIdc::from(100),
+            constraint_flags: ConstraintFlags::from(0),
+            level_idc: 40,
+            seq_parameter_set_id: ParamSetId::from_u32(0).unwrap(),
+            chroma_info: ChromaInfo{
+                chroma_format: ChromaFormat::YUV420,
+                ..ChromaInfo::default()
+            },
+            log2_max_frame_num_minus4: 0,
+            pic_order_cnt: PicOrderCntType::TypeZero {
+                log2_max_pic_order_cnt_lsb_minus4: 2
+            },
+            max_num_ref_frames: 4,
+            gaps_in_frame_num_value_allowed_flag: false,
+            pic_width_in_mbs_minus1: 119,
+            pic_height_in_map_units_minus1: 67,
+            frame_mbs_flags: FrameMbsFlags::Frames,
+            direct_8x8_inference_flag: true,
+            frame_cropping: Some(FrameCropping{
+                bottom_offset: 4,
+                ..FrameCropping::default()
+            }),
+            vui_parameters: Some(VuiParameters{
+                timing_info: Some(TimingInfo{
+                    num_units_in_tick: 1,
+                    time_scale: 60,
+                    fixed_frame_rate_flag: false,
+                }),
+                bitstream_restrictions: Some(BitstreamRestrictions{
+                    motion_vectors_over_pic_boundaries_flag: true,
+                    log2_max_mv_length_horizontal: 11,
+                    log2_max_mv_length_vertical: 11,
+                    max_num_reorder_frames: 2,
+                    max_dec_frame_buffering: 4,
+                    ..BitstreamRestrictions::default()
+                }),
+                ..VuiParameters::default()
+            }),
+        },
+        1920,
+        1080,
+        30.0; "1920x1080 nvidia"
+    )]
+    // This fails.
```

Comment:
Does this failure case represent a bug that needs fixing?  (Be that in this PR or a separate change.)

Reply:
I'm not sure if this case was from a real sample or just used for higher test coverage.

This is the error:
```
`ReaderErrorFor("finish", Error { kind: UnexpectedEof, message: "failed to fill whole buffer" })
```

Removing the comment on the last byte changes the error to this:
```
assertion failed: `(left == right)`
  left: `1084`,
  right: `1080`', h264-reader/src/nal/sps.rs:1619:9
```